docs/configuration.md

```diff
@@ -0,0 +1,541 @@
+Config file: `"baseline": { "provider": 'AzureFileStorage'}`
+
+Sets the storage provider for the baseline used by [with-baseline](#with-baseline-<bool[<:comittish>]>). By default this is set to disk, when the dashboard [reporter](#reporter-<string>) is enabled this is automatically set to Dashboard.
+
+Supported storage providers are:
+
+| Storage location  | Option | Description |
+|------------------ |--------|-------------|
+| Disk              | Disk   | Saves the baseline on disk to the `StrykerOutput` folder |
+| Stryker Dashboard | Dashboard | Saves the baseline to Stryker Dashboard |
+| Azure File Storage | AzureFileStorage | Saves the baseline to Azure File Storage |
+
+For configuring the dashboard provider see [Dashboard Reporter Settings](./reporters.md#dashboard-reporter)
+
+### `baseline.azure-fileshare-url` <`url`>
+
+Default: `null`  
+Command line: `N/A`  
+Config file: `"baseline": { "azure-fileshare-url": 'https://stryker-net.file.core.windows.net/baselines'}`
+
+When using the azure file storage [provider](#baseline.provider-<string>) you must set the file share url.
+The file share url should be in the the format:
+
+`https://<STORAGE_ACCOUNT_NAME>.file.core.windows.net/<FILE_SHARE_NAME>/<OPTIONAL_SUBFOLDER_NAME>`
+
+Providing a subfolder is optional but allowed. The baseline are always stored in a folder called `StrykerOutput/Baselines`. In the case of a custom subfolder the complete url to the baselines would become `https://<FILE_SHARE_URL>/<OPTIONAL_SUBFOLDER_NAME>/StrykerOutput/Baselines`
+
+### `azure-storage-sas` <`string`>
+
+Default: `null`  
+Command line: `--azure-storage-sas "adfdf34343242323rewfe323434"`  
+Config file: `N/A`
+
+When using the azure file storage [provider](#baseline.provider-<string>) you must pass credentials for the fileshare to Stryker.
+For authentication with the azure fileshare we support Shared Access Signatures. For more information on how to configure a SAS check the [Azure documentation](https://docs.microsoft.com/en-us/azure/storage/common/storage-sas-overview).
+
+# Troubleshooting
+
+### `verbosity` <`log-level`>
+
+Default: `info`  
+Command line: `[-V|--verbosity] trace`  
+Config file: `"verbosity": 'trace'`
+
+Change the console `verbosity` of stryker when you want more or less details about the mutation testrun.
+
+All available loglevels are
+* error
+* warning
+* info
+* debug
+* trace
+
+### `log-to-file` <`flag`>
+
+Default: `false`  
+Command line: `[-L|--log-to-file]`  
+Config file: `N/A`
+
+When creating an issue on github you can include a logfile so the issue can be diagnosed easier. 
+
+*\* File logging always uses loglevel `trace`.*
+
+### `dev-mode` <`flag`>
+
+Default: `false`  
+Command line: `--dev-mode`  
+Config file: `N/A`
+
+Stryker will not gracefully recover from compilation errors, instead crash immediately. Used during development to quickly diagnose errors.  
+Also enables more debug logs not generally useful to normal users.
+
+# Misc
```

Comment:
I don't get why these are placed all the way down here. I think they belong with the other baseline options

Reply:
The dashboard api key is used by the dashboard reporter and the dashboard baseline provider. Should it be with reporters or with baseline options?

The dashboard url shouldn't really be used at all, it's more of a development option bit doesn't fit with # Troubleshooting.

So I came up with the Misc/Everything else that doesn't fit somewhere specifically category.

Reply:
The dashboard reporter docs already link to existing docs right? So I'd say we place the API key option with baseline provider. And link from the reporter docs to the baseline provider docs.

Even though the dashboard URL is kinda a development option I still think it should be placed with the baseline options. Maybe put a note with it that you should only use this if you know what you're doing.

Reply:
I disagree completely. It does not make sense for the dashboard api key to be with the baseline options. If anywhere it would be with the dashboard reporter option but that places it out of the way which I don't like. I don't see the harm in having this be it's own little corner. It will show up in the sidebar and it is searchable. We can link from both the dashboard reporter and the baseline provider to this one place instead of hopping from reporter > baseline > api key.